### PR TITLE
refactor: Update efs module

### PR DESCRIPTION
**Pull request by YKharouni**
- Refactor EFS module
- Refactor EFS CSI driver module

### CODE REVIEW

File: storage/aws/efs/main.tf [hunk 1]
```diff
@@ -1,26 +1,40 @@
+# Availability zones
+data "aws_subnet" "private_subnet" {
+  for_each   = toset(var.vpc_cidr_block_private)
+  cidr_block = each.key
+
+  # dummy ternary impose dependency on subnets even though depends_on does not work here
+  vpc_id = var.vpc_subnet_ids != null ? var.vpc_id : var.vpc_id
+}
```
Comment (lemaitre-aneo) [inline storage/aws/efs/main.tf:8]: I think it is much better to give the subnets information directly as input of the module. Re-reading a resource that has been created in the same deployment can lead to strange behavior like some information not propagated upon infrastructure change.

File: storage/aws/efs/variables.tf [hunk 2]
```diff
@@ -5,46 +5,84 @@ variable "tags" {
...
+}
+
+variable "vpc_cidr_block_private" {
+  description = "AWS VPC private cidr block "
+  type        = set(string)
+}
+
+variable "vpc_subnet_ids" {
+  description = "AWS VPC subnet ids"
+  type        = set(string)
```
Comment (lemaitre-aneo) [inline storage/aws/efs/variables.tf:26]: What is the reason behind this change? I think it is much simpler to deal with objects as long as the objects are constructed from other modules.

I would much prefer have this:
```hcl
module "my_vpc" {
  ...
}
module "my_efs" {
  source = ...
  vpc = module.my_vpc
  ...
}
```
rather than this:
```hcl
module "my_vpc" {
  ...
}
module "my_efs" {
  source = ...
  vpc_id = module.my_vpc.id
  vpc_cidr_blocks = module.my_vpc.cidr_blocks
  vpc_cidr_block_private = module.my_vpc.cidr_block_private
  vpc_subnet_ids = module.my_vpc.subnet_ids
  ...
}
```

Of course, this implies to have a well defined module schema.
Although, I have no problem removing the nested structure `efs` as it is the role of this very module to create the EFS.

File: storage/aws/efs/variables.tf [hunk 3]
```diff
@@ -5,46 +5,84 @@ variable "tags" {
...
       "AFTER_90_DAYS"
-    ], var.efs.transition_to_ia)
+    ], var.transition_to_ia)
     error_message = "Possible values for the parameter transition_to_ia are \"AFTER_7_DAYS\" | \"AFTER_14_DAYS\" | \"AFTER_30_DAYS\", \"AFTER_60_DAYS\" | \"AFTER_90_DAYS\"."
   }
 }
+
+variable "access_point" {
+  description = " A map of access point definitions to create"
+  type        = list(string)
```
Comment (lemaitre-aneo) [inline storage/aws/efs/variables.tf:86]:
```suggestion
  type        = set(string)
```

Comment (YKharouni) [reply storage/aws/efs/variables.tf]: Done !

File: kubernetes/aws/addons/efs-csi/examples/complete/main.tf [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+    env             = "test"
+    app             = "complete"
+    module          = "AWS efs-csi"
+    "create by"     = data.aws_caller_identity.current.arn
+    "creation date" = null_resource.timestamp.triggers["creation_date"]
+  }
+}
+
+module "efs_csi" {
+  source                        = "../.."
```
Comment (alouvion-aneo) [inline kubernetes/aws/addons/efs-csi/examples/complete/main.tf:61]: ../folder

**Review by ngruelaneo — DISMISSED** [review]

**Review by ngruelaneo — APPROVED** [review]

**Review by ngruelaneo — APPROVED** [review]
